src/Sylius/Component/Core/Uploader/ImageUploader.php

```diff
@@ -54,7 +54,7 @@ public function upload(ImageInterface $image): void
         do {
             $hash = bin2hex(random_bytes(16));
             $path = $this->expandPath($hash . '.' . $file->guessExtension());
-        } while ($this->filesystem->has($path));
+        } while ($this->filesystem->has($path) || $this->isAdBlockingProne($path));
```

Comment:
We should probably call the new method first, as filesystem calls usually take longer than string lookup.